### PR TITLE
Fix issue #576 Suggestion API not working without max_results

### PR DESCRIPTION
**Pull request by MatonAnthony**
The issue was that the call was passing 0 as a parameter.

### CODE REVIEW

File: app/models/api/suggestions.php [hunk 1]
```diff
@@ -56,6 +56,14 @@
     return $value;
 };
 
+if (empty($get_option('max_results')) || $get_option('max_results') == 0) {
```
Comment (flodolo) [inline app/models/api/suggestions.php]: The right approach is to evaluate `$max_results` inside the called method, not duplicate the call. The function is also very similar to `getTranslationMemoryResults`, so you can take inspiration for the check.

Also note that `$get_option('max_results')` can't really be empty.

**Comment by MatonAnthony** [comment]
I think, it's good this time.

File: app/classes/Transvision/ShowResults.php [hunk 2]
```diff
@@ -58,6 +58,8 @@ public static function getSuggestionsResults($source_strings, $target_strings, $
         ];
         $flat_output = [];
 
+        $max_results = $max_results <= 0 ? 10 : $max_results;
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: Approach is definitely correct. Personally I would swap the check to make it more 'natural' (we want to consider the value provided only if its bigger than 0)
```PHP
$max_results = $max_results > 0 ? $max_results : 10;
```

File: app/classes/Transvision/ShowResults.php [hunk 3]
```diff
@@ -58,6 +58,8 @@ public static function getSuggestionsResults($source_strings, $target_strings, $
         ];
         $flat_output = [];
 
+        $max_results = $max_results > 0 ? $max_results : 10;
```
Comment (pascalchevrel) [inline app/classes/Transvision/ShowResults.php]: I think it would be good to add a comment explaining the issue and why we have to check that $max_results isn't equal to 0.

It would be good to also add a functional test and I think we don't have unit tests for this method.

**Comment by flodolo** [comment]
I think Pascal's request was to create a Unit test (which I didn't add when creating the function, so it's on me). Also, not sure about the functional test added, since we only have 8 results.

File: app/classes/Transvision/ShowResults.php [hunk 4]
```diff
@@ -58,6 +58,9 @@ public static function getSuggestionsResults($source_strings, $target_strings, $
         ];
         $flat_output = [];
 
+        // Make sure $max_results is never set to 0, in which case we set it to 10
```
Comment (flodolo) [inline app/classes/Transvision/ShowResults.php]: I would rephrase this a bit: "Reset $max_results as 10 if it's currently set to 0 (default value when the parameter is not specified in the API request)."

Comment (MatonAnthony) [inline app/classes/Transvision/ShowResults.php]: @flodolo I don't have any idea about what i should be doing for the functional test, it was Pascal request so i did it.

**Comment by flodolo** [comment]
Not sure what happened with all the changes...

**Comment by MatonAnthony** [comment]
We found the issue, it'll be corrected in 5 minutes ;).
Php-cs-fixer went crazy with his config file

**Comment by flodolo** [comment]
I don't think you have any way to fix, besides using master's files and adding back the changes you actually need.

As for the functional tests, I guess it's OK to leave them like this (the fix would be to add more strings), but I would still add some basic unit tests.

**Comment by MatonAnthony** [comment]
Hello, I'll try to fix the first issues tomorrow and to find the time to
write the Unit test
Le 14 févr. 2016 19:55, "Francesco Lodolo" <notifications@github.com> a
écrit :

**Comment by flodolo** [comment]
Note: instead of copying files, you can simply copy the file (or pieces of code) that you want to keep, reset you branch to master, then commit and force push.

**Comment by pascalchevrel** [comment]
Merging, I will do a follow up patch to add a unit test, thanks.
